pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -438,3 +521,27 @@ def is_informative(self):
     def final_transform(self):
         """``True`` if the transform needs to be executed"""
         return self._final_transform
+
+
+@functools.lru_cache
```

Comment:
Do we need an `lru_cache` here? This is only called when we are creating a new transform, and then it gets cached on the transform instance.

Reply:
Good point, I forgot this was here. Another instance of an artifact from the first iteration of the implementation.